packages/shared/testing/evm-based/src/lib/evm-based.custodial.test-factory.ts

```diff
@@ -201,5 +201,33 @@ export const custodialTestFactory = {
         expectHexString(result)
       })
     },
+
+    custodialWalletBatchWithFeesCovered: (
+      sdk: SdkWithCustodialFunctions,
+      testData: BlockchainTestData,
+      chain: 'ETH' | 'ONE' | 'BSC' | 'MATIC' | 'XDC',
```

Comment:
I would use enum values. See https://stackoverflow.com/questions/48850720/typescript-how-can-omit-some-items-from-an-enum-in-typescript

Reply:
That would require a lot of changes all around the sdk, since the final value are union types and through out the sdk Currency is used, etc. Resolving based on discussion.